src/three-components/ARRenderer.ts

```diff
@@ -377,12 +314,13 @@ height: 100%;`);
       return;
     }
 
-    for (const view of (frame as any).views) {
-      const viewport = (session.baseLayer as XRWebGLLayer).getViewport(view);
-      this.renderer!.setViewport(0, 0, viewport.width, viewport.height);
-      this.renderer!.setSize(viewport.width, viewport.height, false);
+    for (const view of (frame as any).getViewerPose(this[$refSpace]).views) {
+      const viewport =
+          (session.renderState.baseLayer as XRWebGLLayer).getViewport(view);
+      this.renderer.setViewport(0, 0, viewport.width, viewport.height);
+      this.renderer.setSize(viewport.width, viewport.height, false);
```

Comment:
setSize seems weird here. Are there ever multiple views? If there were I don't think the size would be the same as the viewport.

Reply:
The "const viewport" assignment above specifically gets the viewport size for use with the current view, either the whole framebuffer or an appropriate subsection. See https://immersive-web.github.io/webxr/#dom-xrwebgllayer-getviewport :
> getViewport() queries the XRViewport the given XRView should use when rendering to the layer.

In this case only one view is expected, but the view loop should work for the more general case also.

Reply:
Sorry, I had described the way it should work, but the implementation didn't match. Should be fixed by https://github.com/GoogleWebComponents/model-viewer/pull/757/commits/311706df0e21924830be4c5c0a7ff18e8ed6171d